src/main/java/stsjorbsmod/patches/DeathPreventionPatch.java

```diff
@@ -13,28 +12,63 @@
 import javassist.CannotCompileException;
 import javassist.expr.ExprEditor;
 import javassist.expr.FieldAccess;
+import stsjorbsmod.cards.DeathPreventionCard;
 import stsjorbsmod.cards.cull.OldBook;
+import stsjorbsmod.cards.cull.ShriekingHat;
 import stsjorbsmod.powers.OldBookPower;
 import stsjorbsmod.util.CardMetaUtils;
 
+import java.util.Comparator;
+import java.util.SortedSet;
+import java.util.TreeSet;
+
 @SpirePatch(clz = DamageInfo.class, method = SpirePatch.CLASS)
-public class OldBookPatch {
-    public static boolean triggerOldBook(AbstractPlayer p, int damageAmount) {
+public class DeathPreventionPatch {
+    public static boolean triggerDeathPrevention(AbstractPlayer p, int damageAmount) {
         if (damageAmount < p.currentHealth)
             return false;
         if (!p.hasPower(OldBookPower.POWER_ID + OldBookPower.UPGRADED) &&
-            !p.hasPower(OldBookPower.POWER_ID + OldBookPower.NORMAL))
+                !p.hasPower(OldBookPower.POWER_ID + OldBookPower.NORMAL) &&
+                p.hand.group.stream().noneMatch(c -> c instanceof ShriekingHat))
             return false;
         if (p.hasPotion(FairyPotion.POTION_ID))
             return false;
-        if (p.hasRelic(LizardTail.ID) && ((LizardTail) p.getRelic(LizardTail.ID)).counter == -1)
+        if (p.hasRelic(LizardTail.ID) && p.getRelic(LizardTail.ID).counter == -1)
             return false;
 
-        OldBookPower po = null;
+        // requested that death preventing cards trigger in order of pickup.
+        SortedSet<DeathPreventionCard> deathPreventionCards = new TreeSet<>(Comparator.comparingInt(DeathPreventionCard::getPriority));
```

Comment:
With the change suggested above, we should be able to loop the `masterDeck` object first to determine "permanent" cards that were added to the deck that need to be consumed, and then pursue the "temporary" cards using your priority implemented.

Reply:
It added a little bit more complexity to the the logic, but it should work as expected.